aws/resource_aws_prometheus_workspace.go

```diff
@@ -0,0 +1,111 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/prometheusservice"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsPrometheusWorkspace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsPrometheusWorkspaceCreate,
+		Read:   resourceAwsPrometheusWorkspaceRead,
+		Update: resourceAwsPrometheusWorkspaceUpdate,
+		Delete: resourceAwsPrometheusWorkspaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"prometheus_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsPrometheusWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Reading AMP workspace %s", d.Id())
+	conn := meta.(*AWSClient).prometheusconn
+
+	details, err := conn.DescribeWorkspace(&prometheusservice.DescribeWorkspaceInput{
+		WorkspaceId: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.Set("arn", aws.StringValue(details.Workspace.Arn))
+	d.Set("prometheus_endpoint", aws.StringValue(details.Workspace.PrometheusEndpoint))
+	d.Set("status", aws.StringValue(details.Workspace.Status.StatusCode))
+	if details.Workspace.Alias != nil {
+		d.Set("alias", aws.StringValue(details.Workspace.Alias))
+	}
```

Comment:
A few items:

* Perform `nil` dereference checking to prevent panics
* `(*schema.ResourceData).Set()` automatically handles pointer values, so no need to wrap fields.
* Recommend skipping the status field unless it is necessary for practitioners.

```suggestion
	if details == nil || details.Workspace == nil {
		return fmt.Errorf("error reading Prometheus Workspace (%s): empty response", d.Id())
	}

	d.Set("alias", details.Workspace.Alias)
	d.Set("arn", details.Workspace.Arn)
	d.Set("prometheus_endpoint", details.Workspace.PrometheusEndpoint)
```

Reply:
changed in upcoming revision